scadalts-ui/src/views/Reports/ReportsPage.vue

```diff
@@ -0,0 +1,941 @@
+			this.reportForm.email= this.selectedReport.email; 
+			this.reportForm.includeData= this.selectedReport.includeData; 
+			this.reportForm.zipData= this.selectedReport.zipData; 
+            this.reportForm.recipients= this.selectedReport.recipients; 
+			
+			this.reportForm.points = this.selectedReport.points.map(x => {
+				const name = this.datapoints.find(p => p.pointId === x.id).name
+				return {...x, name}
+			});
+			if (copy)this.selectedId = -1
+			 
+		},
+		addEmail() {
+			if (!this.validateEmail(this.email)) return
+			if (this.reportForm.recipients.find(x => this.email === x.referenceAddress)) return  
+			this.reportForm.recipients.push({
+				referenceAddress: this.email,
+				referenceType: TYPE_ADDRESS
+			})
+			this.email = ''
+		},
+		addDatapoint({ pointId, name, colour = "", consolidatedChart = true }) {
+			if (this.reportForm.points.find(x => { return x.pointId === pointId})) return
+			this.reportForm.points
+				.push({ pointId, colour, consolidatedChart, name })
+		},
+		async fetchReportList() {
+			this.loading = true;
+			this.reportList = await this.$store.dispatch('fetchReports', {...this.options, keywords: this.search});
+			this.totalReports = this.reportList.total;
+		},
+		async fetchUserList() {
+			this.loading = true;
+			this.userList = [{ id: 0, username: "", email: ''}, ...(await this.$store.dispatch('getAllUsers'))];
+		},
+		runReport(id) {
+			this.$store.dispatch('runReport', id);
+			this.snackbar = true
+			this.snackbarMessage = `${this.$t('reportList.scriptExecuted')} `
+		},
+		updateDateRange() {
+			let bufferDate;
+			if (this.dateRange.endDate < this.dateRange.startDate) {
+				bufferDate = this.dateRange.endDate;
+				this.dateRange.endDate = this.dateRange.startDate;
+				this.dateRange.startDate = bufferDate;
+				this.dateRange.startTime = "00:00";
+				this.dateRange.endTime = "23:59";
+			}
+			if (this.dateRange.startDate) {
+				this.reportForm.fromYear=this.dateRange.startDate.split('-')[0]*1
+				this.reportForm.fromMonth= this.dateRange.startDate.split('-')[1]*1
+				this.reportForm.fromDay= this.dateRange.startDate.split('-')[2]*1
+			}
+
+			if (this.dateRange.startTime) {
+				this.reportForm.fromHour= this.dateRange.startTime.split(':')[0]*1
+				this.reportForm.fromMinute= this.dateRange.startTime.split(':')[1]*1
+			}
+
+			if (this.dateRange.endDate) {
+				this.reportForm.toYear= this.dateRange.endDate.split('-')[0]*1
+				this.reportForm.toMonth= this.dateRange.endDate.split('-')[1]*1
+				this.reportForm.toDay= this.dateRange.endDate.split('-')[2]	*1
+			}
+			if (this.dateRange.endTime) { 
+				this.reportForm.toHour= this.dateRange.endTime.split(':')[0]*1
+				this.reportForm.toMinute= this.dateRange.endTime.split(':')[1]*1
+			}
+		},
+		async saveReport() {
+			if (this.selectedId != -1) {
+				await this.$store.dispatch('saveReport', this.reportForm);
+			} else {
+				await this.$store.dispatch('saveReport', this.reportForm);
+			}
```

Comment:
Why have you using the same `dispatch` method? Maybe you should invoke the "createReport" action on `else` condition? But I'm not sure.